src/overlays/actors/ovl_Lamp_Light/ef_lamp_light.c

```diff
@@ -19,21 +25,114 @@ ActorProfile Lamp_Light_Profile = {
     /* */ Ef_Lamp_Light_actor_ct,
     /* */ Ef_Lamp_Light_actor_dt,
     /* */ Ef_Lamp_Light_actor_move,
-    /* */ (ActorFunc)none_proc1,
+    /* */ (void*)none_proc1,
     /* */ NULL,
 };
-#endif
+static s16 D_80967884_jp[4] = { 2, 2, 1, 0 };
+static s16 D_8096788C_jp[4] = { 0x10, 0x10, 8, 0 };
+static s16* D_80967894_jp[2] = { D_80967884_jp, D_8096788C_jp };
+static s16 D_8096789C_jp[4] = { 0, 0, 0, 0 };
+static s16 D_809678A4_jp[4] = { 0xC8, 0xC8, 0x96, 0 };
+static s16* D_809678AC_jp[2] = { D_8096789C_jp, D_809678A4_jp };
+
+void Ef_Lamp_Light_actor_ct(Actor* thisx, Game_Play* game_play) {
+    s32 pad UNUSED;
+    Lamp_Light* this = (Lamp_Light*)thisx;
+
+    game_play->kankyo.unk_1C = &this->unk_178;
+    Light_diffuse_ct(&this->unk_178, 0, 0x50, 0, (u8)0, (u8)0, (u8)0);
+    this->unk_188 = Global_light_list_new(game_play, &game_play->glight, &this->unk_178);
+}
+
+void Ef_Lamp_Light_actor_dt(Actor* thisx, Game_Play* game_play) {
+    Lamp_Light* this = (Lamp_Light*)thisx;
+
+    game_play->kankyo.unk_1C = NULL;
+    Global_light_list_delete(&game_play->glight, this->unk_188);
+}
+
+s32 func_80967644_jp(void) {
+    s32 var_v1;
+    s32 temp_v0;
+
+    temp_v0 = common_data.time.nowSec;
+    var_v1 = 0;
+    if ((temp_v0 < 0x5460) || (temp_v0 >= 0xFD20)) {
+        var_v1 = 1;
+    }
+    return var_v1;
+}
+
+s32 func_80967678_jp(Lamp_Light* this UNUSED) {
+    s32 var_v1;
+    s32 temp_v0;
+
+    temp_v0 = common_data.time.nowSec;
+    var_v1 = ((temp_v0 >= 0xFD20) && (temp_v0 < 0x14370)) ? 1 : 0;
+    return var_v1;
+}
+
+s32 func_809676BC_jp(void) {
+    s32 var_v1;
+
+    var_v1 = 1;
+    if ((u8)common_data.unk_104AE[0] == 1) {
```

Comment:
This seems to be a case where you need to change the member in `CommonData`
`unk_104AE` currently is an array, but seems like it should be changed to a single member of type `u8`. Don't forget to add extra members to account for the size of the struct